fedimint-core/src/core/server.rs

```diff
@@ -52,6 +52,12 @@ pub trait IServerModule: Debug {
         peer_id: PeerId,
     ) -> anyhow::Result<()>;
 
+    fn verify_input(
+        &self,
+        input: &DynInput,
+        module_instance_id: ModuleInstanceId,
+    ) -> Result<(), DynInputError>;
```

Comment:
Unrelated to the PR since `process_input` does the same: why do we have `module_instance_id` as an argument? `DynInput` already contains it:

```rust
#[doc = " An owned, immutable input to a [`Transaction`](fedimint_core::transaction::Transaction)"]
pub struct DynInput {
    module_instance_id: ::fedimint_core::core::ModuleInstanceId,
    inner: Box<dyn IInput + 'static + Send + Sync>,
}
```

I guess it's a sanity-check and supposed to be set to the *module's* module instance id?  In that case we should probably just use the same macro for `DynServerModule` as for `DynInput`, that includes the instance id in the dyn struct.